### ISSUE TITLE
CameHeader value is wrongly interpreted as BigDecimal and causes ClassCastException

### ISSUE THREAD

**Issue by Jelibo** [apache/camel-kafka-connector#1332]
using `camel-kafka-connector 0.9.0`
using sink connector `CamelAws2s3SinkConnector`

Kafka header `CamelHeader.Id: l1kt74gl9w00` is logged as:
`ConnectHeader(key=CamelHeader.Id, value= l1kt74gl9w00, schema=Schema{STRING})`

Kafka header `CamelHeader.Id: 69203120e780` is logged as:
`ConnectHeader(key=CamelHeader.Id, value=6.9203120E+787, schema=Schema{org.apache.kafka.connect.data.Decimal:BYTES})`

causing the task to fail:

```
org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:614)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:329)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:232)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:201)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:189)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:238)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ClassCastException
```
```
Caused by: java.lang.ClassCastException: class java.math.BigDecimal cannot be cast to class [B (java.math.BigDecimal and [B are in module java.base of loader 'bootstrap')
	at org.apache.camel.kafkaconnector.CamelSinkTask.mapHeader(CamelSinkTask.java:233)
	at org.apache.camel.kafkaconnector.CamelSinkTask.put(CamelSinkTask.java:184)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:586)
	... 10 more
```

A workaround or any suggestion appreciated.

**Comment by valdar** [comment]
Hello, how are the records that the sink consume produced?

It seems that the header does not adhere with the schema declared:
`schema=Schema{org.apache.kafka.connect.data.Decimal:BYTES}` suggest that the value should be a bigdecimal encoded as a byte, so to retrieve it one should do something on the line of: `Decimal.toLogical(schema, (byte[]) header.value())` see [here](https://github.com/apache/camel-kafka-connector/blob/camel-kafka-connector-0.9.0/core/src/main/java/org/apache/camel/kafkaconnector/CamelSinkTask.java#L233). Unfortunately `header.value()` returns a `BigDecimal` instead of a `byte[]` as is declared it should in the provided schema...

**Comment by Jelibo** [comment]
the records are produced by `org.apache.camel.kafkaconnector.rabbitmq.CamelRabbitmqSourceConnector`
there is no fine-grain config per header?

**Comment by valdar** [comment]
> the records are produced by `org.apache.camel.kafkaconnector.rabbitmq.CamelRabbitmqSourceConnector`
> there is no fine-grain config per header?

Which version is this connector?

**Comment by Jelibo** [comment]
Should be the same version `0.9.0`

**Comment by valdar** [comment]
> there is no fine-grain config per header?

What do you mean?

**Comment by valdar** [comment]
After a little bit of investigation, this is actually a bug in how bigdeciaml headers are handled in conjunction with the default `org.apache.kafka.connect.storage.SimpleHeaderConverter`

**Comment by Jelibo** [comment]
> > there is no fine-grain config per header?
> 
> What do you mean?

I meant there is no config option how headers are handled and I have no control over it between these two connectors. I can only disable header by pattern.

**Comment by valdar** [comment]
> > 
> 
> I meant there is no config option how headers are handled and I have no control over it between these two connectors. I can only disable header by pattern.

Well you can plugin a custom HeaderConverter.

It is a bug though, I think I have a fix but releasing it for 0.11.0 and 1.0.0 will take a bit...
